modules/contact/test/tests/explicit_dynamics/tests

```diff
@@ -46,11 +46,42 @@
     type = 'CSVDiff'
     input = 'highvel.i'
     csvdiff = 'highvel_out.csv'
-    abs_zero = 1.0e-4
+    abs_zero = 1.0e-5
     allow_warnings = true
     requirement = 'The system shall be able to solve a simple few-element normal contact problem with contact at an elevated velocity '
                   'using explicit dynamics solving uncoupled, local equations of momentum balance '
                   'during an impact-settling under increased gravity acceleration.'
     heavy = true
   []
+  [exp_constant_mass]
+    type = 'Exodiff'
+    input = 'test_balance.i'
+    exodiff = 'test_balance_out.e'
+    abs_zero = 1.0e-4
+    allow_warnings = true
+    method = '!dbg'
+    cli_args ='Executioner/TimeIntegrator/use_constant_mass=true Mesh/patch_update_strategy=always'
+    requirement = 'The system shall be able to solve an explicit contact problem'
+    'with the constant mass option and with updating the contact patch often.'
+  []
+  [optimized_contact]
+    type = 'Exodiff'
+    custom_cmp = 'optimize.cmp'
+    input = 'test_balance_optimized.i'
+    exodiff = 'test_balance_out.e'
+    abs_zero = 1.0e-4
+    allow_warnings = true
+    method = '!dbg'
+    requirement = 'The system shall be able to solve an explicit contact problem quickly and without extra actions.'
+  []
+  [deep_impact]
+    type = 'Exodiff'
+    input = 'deep_impact.i'
+    exodiff = 'deep_impact_out.e'
+    abs_zero = 1.0e-5
+    allow_warnings = true
+    method = '!dbg'
+    heavy = true
+    requirement = 'The system shall be able to solve for a large displacemnt explicit contact problem'
+  []
```

Comment:
This is a ~7 second test on opt. I think we need this test to make sure we don't regress in exp contact for long duration, deep continued impact. I basically just want the opt version and maybe some parallel test to run this one. No debug, no valgrind, nothing that would time out.